docs/content/docs/data-structures/tree.md

```diff
@@ -567,11 +567,11 @@ For another example, see the `Tree.parent()` method in [Node information](#node-
 
 ### Transactions
 
-If you want the `SharedTree` to treat a set of changes atomically, wrap these changes in a transaction. Using a transaction guarantees that all of the changes will be applied together or none of them will. Transactions will include the concept of a constraint in the future which will allow the changes in a transaction to fail if the constraint is not met.
+If you want the `SharedTree` to treat a set of changes atomically, wrap these changes in a transaction. Using a transaction guarantees that all of the changes will be applied together or none of them will. The Fluid Framework guarantees this already for any sequence of changes that are applied synchronously, in one JS turn, but grouping changes via a transaction when possible allows SharedTree to process them more efficiently. Transactions can also be given constraints, which cause a transaction to be ignored by all clients if the constraint is not met.
```

Comment:
I clarified the language around the "togetherness guarantee" clause (it is only applicable if the transaction is going to apply, which it might not because we have constraints). I wanted to note (as we do in the runTransaction docs) that it's not necessary to use `runTransaction` to have edits be applied synchronously. If we say, "edits in transactions are run synchronously" without the disclaimer, it might seem to imply the inverse as well, but the inverse is not true. FF always guarantees that clients will apply edits synchronously if they were submitted synchronously (because of its batching behavior). Thus, without that, users might think they have to use transactions in scenarios when they actually don't. 

Undo/redo was already mentioned below but I rolled it up into this paragraph, I agree that it flows better that way.

I opted not to talk about intermediate eventing policy here, I agree it's nebulous and maybe even subject to change, and is more the domain of our event libraries and docs anyway.